File: app/src/main/java/com/notai/ui/adapter/TagChipAdapter.java
package com.notai.ui.adapter;

import android.content.res.ColorStateList;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.chip.Chip;
import com.notai.R;
import com.notai.data.model.Tag;

import java.util.HashSet;
import java.util.Set;

/**
 * 标签 Chip 适配器（用于笔记编辑界面）
 */
public class TagChipAdapter extends ListAdapter<Tag, TagChipAdapter.TagChipViewHolder> {
    
    private final Set<Long> selectedTagIds = new HashSet<>();
    private final OnTagSelectedListener onTagSelected;
    
    public interface OnTagSelectedListener {
        void onTagSelected(long tagId, boolean selected);
    }
    
    public TagChipAdapter(OnTagSelectedListener onTagSelected) {
        super(new TagDiffCallback());
        this.onTagSelected = onTagSelected;
    }
    
    public void setSelectedTagIds(Set<Long> ids) {
        selectedTagIds.clear();
        if (ids != null) {
            selectedTagIds.addAll(ids);
        }
        notifyDataSetChanged();
    }
    
    @NonNull
    @Override
    public TagChipViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Chip chip = (Chip) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_tag_chip, parent, false);
        return new TagChipViewHolder(chip);
    }
    
    @Override
    public void onBindViewHolder(@NonNull TagChipViewHolder holder, int position) {
        holder.bind(getItem(position));
    }
    
    class TagChipViewHolder extends RecyclerView.ViewHolder {
        private final Chip chip;
        
        TagChipViewHolder(Chip chip) {
            super(chip);
            this.chip = chip;
        }
        
        void bind(Tag tag) {
            chip.setText(tag.getName());
            chip.setChecked(selectedTagIds.contains(tag.getId()));
            
            try {
                int color = Color.parseColor(tag.getColor());
                chip.setChipBackgroundColor(ColorStateList.valueOf(color));
            } catch (Exception e) {
                chip.setChipBackgroundColor(ColorStateList.valueOf(Color.parseColor("#6200EE")));
            }
            
            chip.setOnCheckedChangeListener((buttonView, isChecked) -> {
                onTagSelected.onTagSelected(tag.getId(), isChecked);
            });
        }
    }
    
    static class TagDiffCallback extends DiffUtil.ItemCallback<Tag> {
        @Override
        public boolean areItemsTheSame(@NonNull Tag oldItem, @NonNull Tag newItem) {
            return oldItem.getId() == newItem.getId();
        }
        
        @Override
        public boolean areContentsTheSame(@NonNull Tag oldItem, @NonNull Tag newItem) {
            return oldItem.equals(newItem);
        }
    }
}

File: app/src/main/java/com/notai/ui/adapter/TagChipAdapter.kt
package com.notai.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.chip.Chip
import com.notai.R
import com.notai.data.model.Tag

/**
 * 标签 Chip 适配器（用于笔记编辑界面）
 */
class TagChipAdapter(
    private val onTagSelected: (Long, Boolean) -> Unit
) : ListAdapter<Tag, TagChipAdapter.TagChipViewHolder>(TagDiffCallback()) {
    
    private val selectedTagIds = mutableSetOf<Long>()
    
    fun setSelectedTagIds(ids: Set<Long>) {
        selectedTagIds.clear()
        selectedTagIds.addAll(ids)
        notifyDataSetChanged()
    }
    
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagChipViewHolder {
        val chip = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_tag_chip, parent, false) as Chip
        return TagChipViewHolder(chip)
    }
    
    override fun onBindViewHolder(holder: TagChipViewHolder, position: Int) {
        holder.bind(getItem(position))
    }
    
    inner class TagChipViewHolder(private val chip: Chip) : RecyclerView.ViewHolder(chip) {
        fun bind(tag: Tag) {
            chip.text = tag.name
            chip.isChecked = selectedTagIds.contains(tag.id)
            
            try {
                val color = android.graphics.Color.parseColor(tag.color)
                chip.chipBackgroundColor = android.content.res.ColorStateList.valueOf(color)
            } catch (e: Exception) {
                chip.chipBackgroundColor = android.content.res.ColorStateList.valueOf(
                    android.graphics.Color.parseColor("#6200EE")
                )
            }
            
            chip.setOnCheckedChangeListener { _, isChecked ->
                onTagSelected(tag.id, isChecked)
            }
        }
    }
    
    class TagDiffCallback : DiffUtil.ItemCallback<Tag>() {
        override fun areItemsTheSame(oldItem: Tag, newItem: Tag): Boolean {
            return oldItem.id == newItem.id
        }
        
        override fun areContentsTheSame(oldItem: Tag, newItem: Tag): Boolean {
            return oldItem == newItem
        }
    }
}

